pkg/asset/ignition/node.go

```diff
@@ -1,44 +1,71 @@
 package ignition
 
 import (
+	"fmt"
 	"path/filepath"
 
-	ignition "github.com/coreos/ignition/config/v2_2/types"
+	"github.com/clarketm/json"
+	"github.com/coreos/ignition/v2/config/util"
+	igntypes "github.com/coreos/ignition/v2/config/v3_1/types"
 	"github.com/vincent-petithory/dataurl"
+	"k8s.io/apimachinery/pkg/runtime"
 
 	"github.com/openshift/installer/pkg/asset"
 )
 
 // FilesFromAsset creates ignition files for each of the files in the specified
 // asset.
-func FilesFromAsset(pathPrefix string, username string, mode int, asset asset.WritableAsset) []ignition.File {
-	var files []ignition.File
+func FilesFromAsset(pathPrefix string, username string, mode int, asset asset.WritableAsset) []igntypes.File {
+	var files []igntypes.File
 	for _, f := range asset.Files() {
 		files = append(files, FileFromBytes(filepath.Join(pathPrefix, f.Filename), username, mode, f.Data))
 	}
 	return files
 }
 
 // FileFromString creates an ignition-config file with the given contents.
-func FileFromString(path string, username string, mode int, contents string) ignition.File {
+func FileFromString(path string, username string, mode int, contents string) igntypes.File {
 	return FileFromBytes(path, username, mode, []byte(contents))
 }
 
 // FileFromBytes creates an ignition-config file with the given contents.
-func FileFromBytes(path string, username string, mode int, contents []byte) ignition.File {
-	return ignition.File{
-		Node: ignition.Node{
-			Filesystem: "root",
-			Path:       path,
-			User: &ignition.NodeUser{
-				Name: username,
+func FileFromBytes(path string, username string, mode int, contents []byte) igntypes.File {
+	return igntypes.File{
+		Node: igntypes.Node{
+			Path: path,
+			User: igntypes.NodeUser{
+				Name: &username,
 			},
+			Overwrite: util.BoolToPtr(true),
 		},
-		FileEmbedded1: ignition.FileEmbedded1{
+		FileEmbedded1: igntypes.FileEmbedded1{
 			Mode: &mode,
-			Contents: ignition.FileContents{
-				Source: dataurl.EncodeBytes(contents),
+			Contents: igntypes.Resource{
+				Source: util.StrToPtr(dataurl.EncodeBytes(contents)),
 			},
 		},
 	}
 }
+
+// ConvertToRawExtension converts and ignition config to a RawExtension containing the ignition as raw bytes
+func ConvertToRawExtension(config igntypes.Config) (runtime.RawExtension, error) {
+	rawIgnConfig, err := json.Marshal(config)
+	if err != nil {
+		return runtime.RawExtension{}, fmt.Errorf("failed to marshal Ignition config: %v", err)
+	}
+
+	return runtime.RawExtension{
+		Raw: rawIgnConfig,
+	}, nil
+}
+
+// ConvertToAppendix converts the contents of an ignition file to an appendix
+func ConvertToAppendix(file *igntypes.File) {
```

Comment:
i'm super clear what this function does, maybe we can add a little more justification? i don't yet understand how appendToFile relates with appendix.. i'm definitely missing something.

Reply:
added an extensive comment - it's a non-trivial change in behaviour